packages/dbos-cloud/register.ts

```diff
@@ -20,26 +20,31 @@ export async function registerUser(username: string, host: string): Promise<numb
     givenName = await prompt("Enter First/Given Name: ");
     familyName = await prompt("Enter Last/Family Name: ");
     company = await prompt("Enter Company: ");
-    const exitCode = await login(username);
-    if (exitCode !== 0) {
-      return exitCode
+    const bearerToken = await authenticate(logger);
+    if (bearerToken === null) {
+      return 1
     }
+    const credentials: DBOSCloudCredentials = {
+      token: bearerToken,
+      userName: username,
+    };
+    writeCredentials(credentials)
   } else {
     const userCredentials = getCloudCredentials();
     if (userCredentials.userName !== username) {
-      logger.error(`You are trying to register ${username}, but are currently logged in as ${userCredentials.userName}. Please run "npx dbos-cloud logout".`)
+      logger.error(`You are trying to register ${username}, but are currently authenticated as ${userCredentials.userName}. Please run "npx dbos-cloud logout".`)
       return 1;
     } else {
-      logger.info(`You are currently logged in as ${userCredentials.userName}.  Registering ${userCredentials.userName} with DBOS Cloud...`)
+      logger.info(`You are currently authenticated as ${userCredentials.userName}.  Registering ${userCredentials.userName} with DBOS Cloud...`)
     }
   }
 
   const userCredentials = getCloudCredentials();
   const bearerToken = "Bearer " + userCredentials.token;
   const loginName = userCredentials.userName;
   try {
-    // First, register the user.
-    const register = await axios.put(
+    // Register the user in DBOS Cloud
```

Comment:
I think we can first check if the user exists -- I went through all login steps and eventually it told me user already exists after done tons of steps. Maybe we can find a way in the future to optimize this experience...